sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java

```diff
@@ -1587,21 +1573,23 @@ public VirtualMachineScaleSetImpl withoutUserAssignedManagedServiceIdentity(Stri
     //
     @Override
     protected void beforeCreating() {
-        if (this.extensions.size() > 0
-            && this.innerModel() != null
-            && this.innerModel().virtualMachineProfile() != null) {
-            this
-                .innerModel()
-                .virtualMachineProfile()
-                .withExtensionProfile(new VirtualMachineScaleSetExtensionProfile())
-                .extensionProfile()
-                .withExtensions(innersFromWrappers(this.extensions.values()));
-        }
+        setExtensions();
     }
 
     @Override
     protected Mono<VirtualMachineScaleSetInner> createInner() {
-        if (isInCreateMode()) {
+        // support flexible vmss with no profile
+        if (this.orchestrationMode() == OrchestrationMode.FLEXIBLE && this.profileNotSet) {
```

Comment:
In update flow, if we get a VMSS with profile, then just call `vmss.update().apply()` (i.e., not update, or maybe only update tags), is this `profileNotSet` true or false?

Or maybe update flow does not get to this method?

Reply:
Marked. Will check it in `update` feature to come.

Reply:
I replaced `profileNotSet` property with a more precise `profileAttached` property. It's an indicator to track vm profile transition from `null` to set. Only if the transition occur will this indicator change to `true` thus profile defaults should be set.
For vm profile null check, in create phase, `sku()` is used as an indicator, while in update phase, `virtualMachineProfile()` itself is used.